actors/eam/src/lib.rs

```diff
@@ -140,8 +140,17 @@ fn create_actor(
     if new_addr.is_reserved() {
         return Err(ActorError::forbidden("cannot create address with a reserved prefix".into()));
     }
+
     let constructor_params =
         RawBytes::serialize(EvmConstructorParams { creator, initcode: initcode.into() })?;
+    let value = rt.message().value_received();
+
+    // Try to resurrect it if it already exists.
+    let f4_addr = Address::new_delegated(EAM_ACTOR_ID, &new_addr.0).unwrap();
+    if let Some(id) = rt.resolve_address(&f4_addr) {
+        rt.send(&Address::new_id(id), METHOD_CONSTRUCTOR, constructor_params.into(), value)?;
+        return Ok(Return { actor_id: id, robust_address: None, eth_address: new_addr });
+    }
```

Comment:
Ooof, moderate pushback here. Re-using the constructor breaks the general convention that the constructor is only called once, upon actor creation in the state tree, and only by the Init actor. I can see three paths forward:

1. Keep this as-is, acknowledging the departure from the pattern, but also acknowledging that the constructor usage is _effectively_ private since no user code can depend on it, therefore easily changeable.
2. Requalify/reterm Constructors to Initializers -- which makes it clearer that they can be called more than once during the lifetime of an actor. However, this is the only case where this happens, so doing this at this stage may generalise a specific circumstance. So I'm not sure we have sufficient motive to do this.
3. Expose a Resurrect public method and dispatch to that. I know you explicitly called out that reusing the Constructor here was simpler, but maybe we just bite the bullet and add the method?

I would feel happier if we chose (3), as we get to keep the convention and IUC, the choice of using the Constructor is implementation simplicity, which I don't think justifies (1) or (2).

Reply:
Yeah, I'll just introduce a new method.